misoc/targets/kasli.py

```diff
@@ -34,71 +36,114 @@ def __init__(self, cd, async_reset):
         ]
 
 
-class _CRG(Module):
-    def __init__(self, platform):
+class _CRG(Module, AutoCSR):
+    def __init__(self, platform, freq=125.0e6):
         self.clock_domains.cd_sys = ClockDomain()
         self.clock_domains.cd_sys4x = ClockDomain(reset_less=True)
         self.clock_domains.cd_sys4x_dqs = ClockDomain(reset_less=True)
         self.clock_domains.cd_clk200 = ClockDomain()
 
+        self.clock_sel = CSRStorage()
+        self.pll_locked = CSRStatus()
+        self.mmcm_locked = CSRStatus()
+
+        # old "sys" clock, now bootstrap
         clk125 = platform.request("clk125_gtp")
         platform.add_period_constraint(clk125, 8.)
         self.clk125_buf = Signal()
-        self.clk125_div2 = Signal()
```

Comment:
ARTIQ is using this. https://github.com/m-labs/artiq/blob/3535d0f1ae98829538b013714aa1a63c0621f9ce/artiq/gateware/targets/kasli.py#L593